Remotion/Validation/UnitTests/RoleCollectors/AddingObjectValidationRuleCollector1Test.cs

```diff
@@ -46,54 +50,159 @@ public void SetUp ()
       _stubObjectValidator1 = new StubObjectValidator();
       _stubObjectValidator2 = new Mock<IObjectValidator>();
       _stubObjectValidator3 = new FakeCustomerValidator();
+      _validatedType = TypeAdapter.Create(typeof(Customer));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void Initialization_CollectorTypeDoesNotImplementIValidationRuleCollector_ThrowsArgumentException ()
     {
+      Assert.That(
+          () => new AddingObjectValidationRuleCollector<Customer>(typeof(Customer)),
+          Throws.ArgumentException
+              .With.ArgumentExceptionMessageEqualTo(
+                  "Parameter 'collectorType' is a 'Remotion.Validation.UnitTests.TestDomain.Customer', "
+                  + "which cannot be assigned to type 'Remotion.Validation.IValidationRuleCollector'.",
+                  "collectorType"));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_InitializesDeferredInitializationValidationMessages ()
     {
+      var validationMessageFactoryStub = new Mock<IValidationMessageFactory>();
+      validationMessageFactoryStub
+          .Setup(e => e.CreateValidationMessageForObjectValidator(It.IsAny<IObjectValidator>(), _validatedType))
+          .Returns(new InvariantValidationMessage("expectedMessage"));
+
+      Func<ObjectValidationRuleInitializationParameters, IObjectValidator> validatorFactory = param => new FakeObjectValidator(param.ValidationMessage);
+
+      _addingObjectValidationRuleCollector.RegisterValidator(validatorFactory);
+      var result = ((IAddingObjectValidationRuleCollector)_addingObjectValidationRuleCollector).CreateValidationRule(validationMessageFactoryStub.Object);
+
+      var fakeObjectValidators = ((ObjectValidationRule<Customer>)result).Validators.Cast<FakeObjectValidator>().ToArray();
+      Assert.That(fakeObjectValidators.Length, Is.EqualTo(1));
```

Comment:
You might want to use `Single()` here, as it throws when there is more than one object contained. Using that, you can reduce these 2 asserts to a single one.

Reply:
This can also be done with all the other ones where you have 2 asserts for a singular object. I don't know if this is actually good here, but I got feedback in my PRs that using Single would be a nice option.

Reply:
Same as above. Thanks for the tip anyway, I will keep this in mind for my next PRs :)